src/components/editor/document-bar/import/import-file.tsx

```diff
@@ -17,19 +17,26 @@ export const ImportFile: React.FC<ImportProps> = ({ noteContent, updateNoteConte
       }
       const file = fileInput.files[0]
       const fileReader = new FileReader()
-      fileReader.readAsText(file)
       fileReader.addEventListener('load', () => {
         const newContent = fileReader.result as string
-        updateNoteContent(noteContent + '\n' + newContent)
+        if (noteContent.length === 0) {
+          updateNoteContent(newContent)
+        } else {
+          updateNoteContent(noteContent + '\n' + newContent)
+        }
```

Comment:
should an upload really be appended? I thought it should replace the current content and create a new revision.

Reply:
I would differentiate between the API call `PUT /note ` and the editor UI. Afaik the other imports don't replace the whole note content either.